src/gossip/service.zig

```diff
@@ -1248,7 +1272,7 @@ pub const GossipService = struct {
         done: Atomic(bool) = Atomic(bool).init(false),
 
         pub fn deinit(this: *PullRequestTask) void {
-            if (!this.output_consumed.load(.acquire)) {
+            if (this.output_consumed.load(.acquire)) {
```

Comment:
yeah - output will always be consumed in the end of the method (the only place this struct is used rn) - still modified the deinit to throw an error to be extra safe

138064b0

Reply:
Returning an error from a deinit function feels kinda wrong.

I don't understand the purpose of checking `output_consumed`. Why wouldn't the deinit function simply deinit the arraylist unconditionally? Normally I would expect a deinit function to conclusively deinit something.

If you want to validate that the output has been consumed, why not put an assertion in the defer that the output was consumed, instead of the `unreachable?`